packages/db-model/src/model.ts

```diff
@@ -74,84 +75,123 @@ export function createBaseModel(resolver?: ConnectionResolver) {
     /**
      * Generic query builder.
      *
-     * @param {(qb: Knex | Transaction) => QueryBuilder} callback
      * @param {Transaction} [trx]
-     * @returns {Promise<T[]>}
+     * @returns {(Knex.Transaction | Knex)}
      */
-    public static async buildQuery<T>(
-      callback: (qb: Knex | Transaction) => QueryBuilder,
-      trx?: Transaction
-    ): Promise<T[]> {
-      const qb = db.queryBuilder(this.getConnection(), trx);
-      const result = await callback(qb);
-
-      return object.toCamelCase<T[]>(result);
+    public static queryBuilder(trx?: Transaction): Knex.Transaction | Knex {
+      return db.queryBuilder(this.getConnection(), trx);
     }
 
     /**
      * Finds a record based on the params.
-     * Returns null if no results were found.
      *
      * @param {object} [params={}]
+     * @param {Function} callback
+     * @param {Knex.Transaction} trx
+     * @returns {Knex.QueryBuilder}
+     */
+    public static find<T>(params: object = {}, callback?: any, trx?: Knex.Transaction): Knex.QueryBuilder {
+      return db.find<T>(this.getConnection(), this.table, params, callback, trx);
+    }
+
+    /**
+     * Finds the first record based on the params.
+     *
+     * @param {object} [params={}]
+     * @param {Function} callback
      * @param {Knex.Transaction} trx
      * @returns {Promise<T | null>}
      */
-    public static get<T>(params: object = {}, trx?: Knex.Transaction): Promise<T | null> {
-      return db.get<T>(this.getConnection(), this.table, params, trx);
+    public static async findFirst<T>(params: object = {}, callback?: any, trx?: Knex.Transaction): Promise<T | null> {
+      return db.findFirst<T>(this.getConnection(), this.table, params, any, trx);
     }
 
     /**
      * Find record by it's id.
-     * Returns null if not found.
+     * Throws an exception if not found.
      *
-     * @param {number} id
+     * @param {any} id
+     * @param {Function} callback
      * @param {Knex.Transaction} trx
      * @returns {Promise<T | null>}
      */
-    public static getById<T>(id: number, trx?: Knex.Transaction): Promise<T | null> {
-      return db.getById<T>(this.getConnection(), this.table, id, trx);
+    public static findById<T>(id: any, callback?: any, trx?: Knex.Transaction): Promise<T | null> {
```

Comment:
Same here - we won't need callback.

Reply:
@FriedDust Yes, why do we need a callback?

Reply:
I think we can make improvements after the release.